DragonSlayer.cpp

```diff
@@ -1,19 +1,26 @@
 #include "DragonSlayer.h"
 #include "Dragon.h"
+#include "Utility.h"
 
-//DragonSlayer::DragonSlayer
+DragonSlayer::DragonSlayer( const std::string& name_, int hp, int armor ) : Character(hp, armor, 4), name(name_)
+{
+    this->generateItems();
+}
 
-//DragonSlayer::getName
+const std::string& DragonSlayer::getName()
+{
+    return this->name;
+}
 
 void DragonSlayer::attack(Character& other)
 {
     std::cout << name << " is attacking " << other.getName() << " !!" << std::endl;
     if( auto* dragon = dynamic_cast<Dragon*>(&other) )
     {
-        assert(false);
         //DragonSlayers get a 10x boost when attacking dragons, from their attack item.
         //so they should USE their attack item before attacking the dragon... 
-        //
+        attackItem->use(this);
```

Comment:
- [ ] FIXME: you need to reset after use, since these are 1-time-use only items.